### ISSUE TITLE
JOSS Review: User-Friendly Summary of API Documentation

### ISSUE THREAD

**Issue by rrjbca** [athob/py-ananke#7]
The authors have documented the core Ananke class, it's methods and arguments to a high level using docstrings and an accompanying example in a Jupyter notebook. However, all of this documentation exists within the code and is spread across multiple files making it somewhat difficult for a new user to find and read. 

I would suggest that the authors start a document named "Documentation" in a user-friendly format (e.g. markdown) and clearly linked from the readme. It should present a brief summary of the relevant API level documentation in one place and in a format that is accessible for users, with instructions/links for where to find further information. As a minimum I believe the document should include the following:

- [x] A high-level description of using py-ananke (i.e. "Create an `Ananke` pipeline object and call `run` to return a Galaxia `Output` survey")
- [x] A link to the jupyter notebook example
- [x] Instructions to view the `Ananke` class constructor docstring (i.e. "run `help(Ananke.__init__)`")
- [x] State the expected format of the `particles` dictionary (as it is in the example)
- [x] Instructions to view the `d_params`, `e_params` and `err_params` options (i.e. "call `Ananke.display_density_docs()`" etc)
- [x] Instructions to view the `Ananke.run` docstring (i.e."run `help(Ananke.run)`")
- [x] A list of the additional `kwargs` that can be passed to  `Ananke.run`
- [x] Description of the `Galaxia.Output` data structure in which the output data is returned

If the authors wished to use a more advanced tool like sphinx to incorporate docstrings directly into the documentation that would be excellent but I recognize that would be significantly more work to setup and I'm happy for it to be deferred for future work.

**Comment by athob** [comment]
Thank you for your feedback. In order to make use of the existing set of docstrings, and as a starting point to put together an appropriate documentation, I've chosen to use sphinx and readthedocs. The [latest release](../tree/8769c5d2a8b565ada26a6485b660969b6765c270) has the corresponding documentation source incorporated in the repository in the `docs` directory. I also added the readthedocs link in the "About" section for ease of access.

With that done, I will be including each of the items you mentioned and comment back once it's all done.

**Comment by athob** [comment]
I continued working on this issue in the dedicated [7-joss-review-api-doc](../tree/7-joss-review-api-doc) branch. I have also activated the readthedocs autobuild for that branch specifically. This documentation is available at this link https://py-ananke.readthedocs.io/en/7-joss-review-api-doc/.

Lately, I have improved the sphinx workflow so that it can make better use of the existing README on the main github page. An issue I notably had before was related to my use of internal links. It was notably problematic as the example notebook was referred in the README via such internal link, which didn't render well in the sphinx-generated readthedocs.

Since I wanted that example notebook to be the main resource for user to first get their hand on the software, I aimed to include the points you mentioned in it. After fixing the sphinx generation, and the internal link rendering, I improved the existing notebook to reflect those points. I have yet to list the additional `kwargs` that can be passed to `Ananke.run`, I'll comment once that is done.

Note that I am having some other rendering issues for which I have not found a solution yet. It notably relates to the formatting of the output cells, which I haven't a way to make scrollable (for readability), whether in the case of pure text outputs, or html-formatted vaex tables.

I also noticed that the [furo theme](https://sphinx-themes.org/sample-sites/furo/) I was using had poor choice of color in dark mode for the latter tables: for this, I temporarily switched to the default readthedocs theme until I can find a way to use furo.

**Comment by athob** [comment]
In [7-joss-review-api-doc](../tree/7-joss-review-api-doc)'s last few updates, I was able to improve the `Ananke` docstrings in a way that it now dynamically uses the `Galaxia_ananke` docstrings to carry over specifically selected parameters descriptions. This allowed me to also complete the documentation of the `Ananke.run` method, as well as properly identifying the parameters to list in the example notebook.

I also found a better sphinx theme (the [pydata-sphinx-theme](https://sphinx-themes.org/sample-sites/pydata-sphinx-theme)) that allowed better dark mode support than furo while also enabling the horizontal scrolling I needed for the html-formatted vaex tables.

I believe that with this last updates, this completes all the tasks that were listed.
